internal/handlers/initrd.go

```diff
@@ -89,6 +92,22 @@ func initrdOverlayReader(imageStore imagestore.ImageStore, client *AssistedServi
 			return nil, "", http.StatusInternalServerError, fmt.Errorf("failed to create append reader for initrd: %v", err)
 		}
 
+		versionOK, err := common.VersionGreaterOrEqual(version, isoeditor.MinimalVersionForNmstatectl)
+		if err != nil {
+			return nil, "", http.StatusInternalServerError, err
+		}
+
+		if versionOK {
+			nmstatectlPath := imageStore.NmstatectlPathForParams(version, arch)
+			nmstateImgContent, err := os.ReadFile(nmstatectlPath)
+			if err != nil {
+				return nil, "", http.StatusInternalServerError, fmt.Errorf("failed to read nmstate img: %v", err)
+			}
+			initrdReader, err = overlay.NewAppendReader(initrdReader, bytes.NewReader(nmstateImgContent))
```

Comment:
There's no reason to read the content into memory then create a reader for the content.
`os.Open(nmstatectlPath)` should give you a reader that can then be passed to `NewAppendReader`